components/sup/src/manager/service/mod.rs

```diff
@@ -98,45 +102,32 @@ pub struct Service {
 }
 
 impl Service {
-    fn new(local_member_id: MemberId,
+    fn new(sys: Arc<Sys>,
            package: PackageInstall,
            spec: ServiceSpec,
-           gossip_listen: &GossipListenAddr,
-           http_listen: &http_gateway::ListenAddr,
            manager_fs_cfg: Arc<manager::FsCfg>,
            organization: Option<&str>)
            -> Result<Service> {
         spec.validate(&package)?;
+        let pkg = Pkg::from_install(package)?;
         let spec_file = manager_fs_cfg.specs_path.join(spec.file_name());
-        let service_group = ServiceGroup::new(&package.ident().name, spec.group, organization)?;
-        let runtime_cfg = Self::runtime_config_from(&package)?;
-        let config_root = spec.config_from
-            .clone()
-            .unwrap_or(package.installed_path.clone());
-        let svc_cfg = ServiceConfig::new(&package,
-                                         &runtime_cfg,
-                                         config_root,
-                                         spec.binds.clone(),
-                                         &gossip_listen,
-                                         &http_listen)?;
-        let hook_template_path = svc_cfg.config_root.join("hooks");
-        let hooks_path = fs::svc_hooks_path(service_group.service());
-        let locked_package = Arc::new(RwLock::new(package));
+        let service_group = ServiceGroup::new(&pkg.name, spec.group, organization)?;
+        let config_root = Self::config_root(&pkg, spec.config_from.as_ref().map(|p| &**p));
```

Comment:
Aha. For me this was not evident. If there is a cleaner way to express this I'm for it.